xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultModelBridge.java

```diff
@@ -439,7 +439,9 @@ public boolean restoreDeletedDocument(long deletedDocumentId, boolean checkConte
                 // Default for now is to skip and log as error to restore over existing documents.
                 logger.error("Document [{}] with ID [{}] can not be restored. Document already exists",
                     deletedDocument.getFullName(), deletedDocumentId);
-            } else if (checkContextUser && !canRestoreDeletedDocument(deletedDocumentId, context.getUserReference())) {
+            } else if (checkContextUser && (!canRestoreDeletedDocument(deletedDocumentId, context.getUserReference())
```

Comment:
Would probably be better to have two different message depending if the one who does not have the right is the user or the author. Otherwise you end up with "WTF I am admin !".

Reply:
Right.